module-2/backend/week_two.md

```diff
@@ -26,29 +34,81 @@ end
 
 Now how would you find the owner of the team with an id of 4?
 
-3. What do they allow you to do?
-7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
-8. Define foreign key, primary key, and schema.
+3. What do they allow you to do? 
+      Team.find(id).owner_id
+7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram. 
+      Students        Teachers        Students_Teachers
+      name            name            student_id
+                                      teacher_id
+8. Define foreign key, primary key, and schema. 
+      Primary key - the inherant id of a table
+      Foreign key - the id of another table 
+      Schema - the state of a database
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
-10. What are the parts of an HTTP response?
-11. Describe some techniques to make our Sinatra code more DRY. Give an example of when you would use these techniques.
+      The foreign key is the repesentation of that primary key but on another table 
+10. What are the parts of an HTTP response? 
+      1. First the user makes a request (some instance of CRUD) on that website
+      2. That request is given to the controller which detirmines what the user was looking for and prepares all needed 
+      responses
+      3. To prepare the response the controller asks the model for attr/methods 
+      4. The model prepares anything it needs and asks the database for any specific info it needs to complete the methods, or
+      details of different attributes.  
+      5. The database response with the details to give back to the model
+      6. The model response to the controller with necessary attributes/methods
+      7. The controller then response to the user and shows a desired view they requested, with the correct info from the 
```

Comment:
This is a pretty solid description of the MVC pattern. May have included another line with the controller going to find the view and populate it with the information it collected from the model.

However! We were asking about the parts of an HTTP response, which is just the headers and body (also nice to know that the status code is the first line of the headers). You went deeper to describe how that response is created.